### DISCUSSION TITLE
How to do createMany or a transaction where I can create many records?

### DISCUSSION THREAD

**Discussion by PhilAndrew** [MichalLytek/typegraphql-prisma#123] [Q&A] [answered]
I need to import 1 million records into my database.
Creating the record one-by-one isn't good as it commits the record each time and really that's too slow, limits the number of inserts I can do in a second.

I would like a createMany function or the ability to do a long transaction.

To work-around this issue I had to use a different library other than this and Prisma.

    const db = await open({
        filename: './prisma/dev.db',
        driver: sqlite3.Database
    })
    const result0 = await db.exec('DELETE FROM Book where 1=1')
    const result1 = await db.exec('BEGIN TRANSACTION')
            for (const d of dataIs) {
                const result = await db.run('INSERT INTO Book(id, title, author) VALUES (:id, :title, :author)', {
                    ':id': d.id,
                    ':title': d.title,
                    ':author': d.author,
                })
            }
    const result = await db.exec('END TRANSACTION')
    await db.close()

Can anyone suggest how to do with this?

**Comment by MichalLytek** [comment, marked as answer]
I think `createMany` is supported since Prisma 2.20 🤔

**Reply by PhilAndrew** [reply]
Well.... it doesn't appear to be in the generated code, 

`import { Book } from "@generated/type-graphql/models/Book";`

I see

```
  export type PrismaAction =
    | 'findUnique'
    | 'findMany'
    | 'findFirst'
    | 'create'
    | 'createMany'
```

but.. there's no createMany function in that Book.ts

**Reply by MichalLytek** [reply]
> Remarks
> createMany is not supported by SQLite.

**Reply by PhilAndrew** [reply]
ahhhhhh.. sad feeling
